cmd/format-erasure.go

```diff
@@ -62,12 +61,8 @@ const offlineDiskUUID = "ffffffff-ffff-ffff-ffff-ffffffffffff"
 var formatHealErrors = map[error]struct{}{
 	errUnformattedDisk: {},
 	errDiskNotFound:    {},
-}
-
-// List of errors considered critical for disk formatting.
-var formatCriticalErrors = map[error]struct{}{
-	errCorruptedFormat: {},
 	errFaultyDisk:      {},
+	errCorruptedFormat: {},
```

Comment:
Not sure why we are adding errFaultyDisk and errCorruptedFormat to formatHealErrors, which represents, the list of errors that healing can fix.

Reply:
@vadmeste yes because there are room and opportunity for healing to trigger writes to see if we can 

- in case of faulty disks write to new block location on the drive which is still healthy to maintain quorum
- in case of corrupted format, it is `xl.meta` which not readable or corrupt

Reply:
In-case of disk format, these are simply ignored because `format` is nil - it is to provide tolerance towards 
such disks just like errDiskNotFound so that we ignore those drives but we let the cluster be operational 

That is the crux of what this PR is trying to fix @vadmeste making sure that the errors such as errFaultyDisk and errCorruptedFormat are not that catastrophic - instead, they are simply disks to be ignored.